### PR TITLE
Issue #11241: fix empty line separator for enum and interface fields

### LINKED ISSUE #11241: `EmptyLineSeparator`: false positive in Enum
I have read check documentation: https://checkstyle.org/config_whitespace.html#EmptyLineSeparator
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac SomeEnum.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="EmptyLineSeparator">
            <property name="allowMultipleEmptyLines" value="false"/>
            <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
        </module>
    </module>
</module>

/var/tmp $ cat SomeEnum.java
package some.pckg;

import java.util.function.Function;

public enum SomeEnum {

    ONE("one"),
    TWO("two");




    private final String someString;

    SomeEnum(String someString) {
        this.someString = someString;
    }

    public String getSomeString() {
        return someString;
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2.1-all.jar -c config.xml SomeEnum.java
Checkstyle ends without errors.
```

---

Checkstyle ends without errors, but I expect violation to be found since there are more than 1 empty lines between enum constant `TWO` and filed `someString`.

### PR DESCRIPTION
**Pull request by strkkk**
Closes #11241 

for some reason, check was not aware that enums can have fields

Diff Regression config: https://gist.githubusercontent.com/strkkk/31749e0743553a606f69fdfc6670d780/raw/332b1574376419f7ea927be6902398aedec8e3bf/confg.xml

### CODE REVIEW

**Comment by strkkk** [comment]
Github, generate report

**Comment by strkkk** [comment]
pgjdbc failure is valid, issue has to be fixed in their repo. https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/557385672

**Comment by strkkk** [comment]
it looks like that report takes too much time, I will try to narrow projects list to speed it up

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java [hunk 1]
```diff
@@ -936,7 +936,7 @@ private static boolean isPrecededByJavadoc(DetailAST token) {
      */
     private static boolean isTypeField(DetailAST variableDef) {
         return TokenUtil.isOfType(variableDef.getParent().getParent(),
```
Comment (nmancus1) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java:938]: Interfaces can have fields,too (if they should is another discussion). Can we just use TokenUtil#isTypeDeclaration here?

**Review by nmancus1 — CHANGES_REQUESTED** [review]
One item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java]: yes, you are right. I fixed this

**Comment by strkkk** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java (hunk 1 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorEnumMembers.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptylineseparator;
+
+enum InputEmptyLineSeparatorEnumMembers {
+
+    ONE("one"),
+    TWO("two");
+
+
+    private final String someString; // violation
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorEnumMembers.java:21]: Please add violation message to comment, this check has four different messages.

**Review by nmancus1 — CHANGES_REQUESTED** [review]
pgjdbc PR is merged.

Also, one minor:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorEnumMembers.java (hunk 2 above)
Comment (strkkk) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorEnumMembers.java]: done

**Comment by strkkk** [comment]
@nmancus1 for some reason equalsverifier failed with tons of errors that are not related with my changes - https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/557583409 

may be they messed up something in their project since we use their latest master

**Comment by romani** [comment]
Github, rebase

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorEnumMembers.java (hunk 2 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorEnumMembers.java]: Resolved.

**Review by nmancus1 — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Review by romani — CHANGES_REQUESTED** [review]
Do you know why pitest is missing ?

Please add to input:
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9443c60_2022151858/reports/diff/Vavr/index.html#A1

**Comment by strkkk** [comment]
> Please add to input:

@romani done

**Comment by romani** [comment]
To input - New violation when  no space between enum fields
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9443c60_2022151858/reports/diff/Vavr/index.html#A6

**Comment by strkkk** [comment]
> To input - New violation when no space between enum fields 

@romani added

> Do you know why pitest is missing ?

It is in ci list now, I did local rebase so it may be a reason

**Comment by romani** [comment]
new violation to input - https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9443c60_2022151858/reports/diff/Orekit/index.html#A17 static block is not separated.

**Comment by strkkk** [comment]
@romani done

**Review by romani — APPROVED** [review]

**Pull request merged**